dozer/cogs/voice.py

```diff
@@ -27,6 +28,64 @@ async def on_voice_state_update(self, member, before, after):
                 if len(config) != 0:
                     await member.add_roles(member.guild.get_role(config[0].role_id))
 
+    @Cog.listener('on_voice_state_update')  # Used for auto PTT
+    async def on_PTT_check(self, member, before, after):
+        """Handles voice activity when members join/leave voice channels"""
+        # skip this if we have no perms to edit voice channel
+        total_users = 0
+        if member.guild.me.guild_permissions.manage_channels and before.channel != after.channel:
+            # determine if it's a join/leave event as well.
+            # before and after are voice states
+            if before.channel is not None:
+                # leave event, take role
+                voice_channel = before.channel
+                total_users = len(before.channel.members)
+                config = await AutoPTT.get_by(channel_id=before.channel.id)
+            if after.channel is not None:
+                # join event, give role
+                voice_channel = after.channel
+                total_users = len(after.channel.members)
+                config = await AutoPTT.get_by(channel_id=after.channel.id)
+
+            everyone = voice_channel.guild.default_role  # grab the @everyone role
+            perms = voice_channel.overwrites_for(everyone)  # grab the @everyone overwrites
+
+            if total_users > config[0].ptt_limit:
+                perms.update(use_voice_activation=False)  # Force PTT enable
+            if total_users <= config[0].ptt_limit:
+                perms.update(use_voice_activation=None)  # Set PTT to neutral
+            await voice_channel.set_permissions(target=everyone, overwrite=perms)
+
+    @command()
+    @bot_has_permissions(manage_channels=True)
+    @has_permissions(manage_channels=True)
+    async def autoptt(self, ctx, voice_channel: discord.VoiceChannel, ptt_threshold: int):
+        """Configures AutoPtt limit for when members join/leave voice channels ptt is enabled"""
+        config = await AutoPTT.get_by(channel_id=voice_channel.id)
+        if len(config) != 0:
+            await AutoPTT.delete(id=config[0].id)
```

Comment:
If auto-PTT is set for a channel and then set to 0/a negative number, the previous config would be erased. This should be moved below the following block. Or, if the channel ID is used as the primary key (see further down), this could be removed and `update_or_add` will take care of the `UPDATE` if needed.